lua/orgmode/org/mappings.lua

```diff
@@ -394,9 +397,53 @@ function OrgMappings:_todo_change_state(direction)
     return dispatchEvent()
   end
 
-  local log_note = config.org_log_done == 'note'
-  local log_time = config.org_log_done == 'time'
-  local should_log_time = log_note or log_time
+  local new_state = item.todo_keyword.value
+
+  -- Determine which configuration to use
+  local global_config = config.org_log_done
+
+  --- @type nil | false | 'time' | 'note'
+  local section_config
+  local logging_prop = item.properties.items.logging
+  if logging_prop == 'nil' then
+    section_config = false
+  elseif logging_prop ~= nil then
```

Comment:
Having headline or file scoped config for todo keywords requires many more changes across the codebase to be properly considered. It will work on state change, but agenda, highlights, etc. will not work as expected.

Can we have this PR only handle `@` and `!` in the todo keywords and leave the file/headline level todo keywords for a different PR?

Reply:
I reverted the 2 commits which targeted the mappings file. If I remember right, this shouldn't affected parsing the "global" config value, but I will reconfirm this with my personal usage tomorrow.